src/http-client.js

```diff
@@ -104,14 +120,26 @@ export class HttpClient {
   fetch(input: Request|string, init?: RequestInit): Promise<Response> {
     this::trackRequestStart();
 
-    let request = Promise.resolve().then(() => this::buildRequest(input, init, this.defaults));
+    let request = Promise.resolve().then(() => this::buildRequest(input, init, this.defaults))
+      .then( result => {
+        const { retryConfig } = result;
+        if (retryConfig) {
+          result.retryConfig = retryConfig;
+        }
+        return result;
+      });
```

Comment:
This belongs in the interceptor. The HttpClient should know as little as possible about any interceptors.

Reply:
I need to sit with this coffee for a bit to figure out why it is here, as it doesn't look necessary at all as I look at it right now.

Reply:
I've deleted this code. This seems to be vestigial code.